### ISSUE TITLE
Feature request: Expose app version in Apps SDK context

### ISSUE THREAD

**Issue by steffenloesch** [OfficeDev/microsoft-teams-library-js#1731]
In some cases, the MetaOS app upgrade flow requires explicit upgrade consent from a user to upgrade an app. So app developers can't be sure that users are on the latest version of their app. At the same time the app version is not exposed to the app itself, so app developers also don't have insight in what version they're actually running on. Knowing the version would help with debugging, data analysis, and adjusting the app's behavior based on the version (as capabilities could vary with the version).

**Comment by Prasad-MSFT** [comment]
@steffenloesch - Thanks for bringing this up. We will pass on the feature request to engineering team.

**Comment by nwojod-MSFT** [comment]
@steffenloesch - Thanks for raising this request and we believe this would be a good feature for developers. We have added it to our backlog for tracking. The team will review it, the level of work required, and we will prioritize it accordingly. We are unable to promise when this feature will be implemented. We will provide an estimate on this thread as soon as we have a committed timeline.

**Comment by AE-MS** [comment]
@steffenloesch to be sure I understand the request, when you say that you would like to know the app version, do you mean the [version that has been put into the manifest](https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema#version)?

**Comment by steffenloesch** [comment]
@AE-MS yes I mean the app version in the manifest (not the version of the Teams client itself). So what you've linked is exactly what I mean.

**Comment by AE-MS** [comment]
Great, thanks for the clarification. We'll look into adding something to help with this.

As an intermediate workaround, depending on how you have your build system setup, you could have the app version number defined in one place in your project, then "stamped" into both the manifest and as a constant in your code somewhere (so you have it available at runtime) using a pre-processing step in your build process. Not sure if this would solve the exact scenario you are dealing with (or if you have a build system setup to allow this), but thought I'd offer it as a suggestion in case it worked for you.

**Comment by steffenloesch** [comment]
Sounds good, thanks for providing ideas for a workaround! As an alternative, I think I could also set the version as query string parameter in the URLs of the manifest, and pass it to the app that way. That way I'd only need manifest changes. Nevertheless, it would be nicer if this was exposed in the SDK, so thanks for looking into it!
